src/Interpreters/ProfileEventsExt.cpp

```diff
@@ -86,9 +86,16 @@ static void dumpMemoryTracker(ProfileEventsSnapshot const & snapshot, DB::Mutabl
     columns[i++]->insert(static_cast<UInt64>(snapshot.current_time));
     columns[i++]->insert(static_cast<UInt64>(snapshot.thread_id));
     columns[i++]->insert(Type::GAUGE);
-
     columns[i++]->insertData(MemoryTracker::USAGE_EVENT_NAME, strlen(MemoryTracker::USAGE_EVENT_NAME));
-    columns[i++]->insert(snapshot.memory_usage);
+    columns[i]->insert(snapshot.memory_usage);
+
+    i = 0;
+    columns[i++]->insertData(host_name.data(), host_name.size());
+    columns[i++]->insert(static_cast<UInt64>(snapshot.current_time));
+    columns[i++]->insert(static_cast<UInt64>(snapshot.thread_id));
+    columns[i++]->insert(Type::GAUGE);
+    columns[i++]->insertData(MemoryTracker::PEAK_USAGE_EVENT_NAME, strlen(MemoryTracker::PEAK_USAGE_EVENT_NAME));
+    columns[i]->insert(snapshot.peak_memory_usage);
 }
```

Comment:
Please explain those lines to me. I do not understand it.

Reply:
The data (ProfileEvents and memory events) between the server and the client passes through row in the sql block (DB::Block). I added one more row with my data (peak memory). 
Columns:
```
static const NamesAndTypesList column_names_and_types = {
        {"host_name", std::make_shared<DataTypeString>()},
        {"current_time", std::make_shared<DataTypeDateTime>()},
        {"thread_id", std::make_shared<DataTypeUInt64>()},
        {"type", TypeEnum},
        {"name", std::make_shared<DataTypeString>()},
        {"value", std::make_shared<DataTypeInt64>()},
    };
```

Reply:
That is Ok.
The diff itself looks strange.

![image](https://github.com/ClickHouse/ClickHouse/assets/104093494/7e4b0351-0f9d-49ad-acc9-10d36c62bc51)

Looks like one write is not enough and you added two :)

Reply:
Oh, my bad. 
That is right. You add one more row here.

Reply:
I don't understand what exactly is confusing? The first block write current_memory_usage. Insert() `appends new value at the end of column (column's size is increased by 1).`  i++ in block ends is not necessary, i replaced to i;